SimCalorimetry/EcalTestBeam/interface/EcalTBDigiProducer.h

```diff
@@ -35,7 +37,7 @@ class EcalTBDigiProducer : public EcalDigiProducer {
   void fillTBTDCRawInfo(EcalTBTDCRawInfo &theTBTDCRawInfo);
 
   const EcalTrigTowerConstituentsMap m_theTTmap;
```

Comment:
Is the `m_theTTmap` initialised somewhere?

Reply:
No apparently not. I think the code inherits from full ECAL code where trigger tower information was needed. Here an empy map has been used by the analyzers of the ECAL TB.

Reply:
But as far as I can see that map is needed by the `performReadout` method of EcalTBReadout: am I wrong?
How can this code work as such? Is it used? Please let me know if I misunderstood how does it work.

If this code is obsolete, better to remove it completely rather than having to spend time in maintaining it. Otherwise, it should be fixed, profiting of this PR

Reply:
I shall be against removing this code. The test beam data were always useful in judging the quality of Geant4 physics. I shall refer this to Fabio Cossutti and Chiara Rovelli who could evaluate the utility of this code. I think they were involved in the analysis of H4 test beam data.

Reply:
> I shall be against removing this code. The test beam data were always useful in judging the quality of Geant4 physics. I shall refer this to Fabio Cossutti and Chiara Rovelli who could evaluate the utility of this code. I think they were involved in the analysis of H4 test beam data.

I fully agree @bsunanda 
Please provide the needed fix/patch here to allow it being able to run, then
@cms-sw/ecal-dpg-l2 is expected to provide help for it, I'd say